frontend/packages/citizen-frontend/src/applications/editor/FileUpload.tsx

```diff
@@ -0,0 +1,342 @@
+import React, { useState } from 'react'
+import styled from 'styled-components'
+import { useTranslation } from '~localization'
+import { Gap } from '@evaka/lib-components/src/white-space'
+import colors from '@evaka/lib-components/src/colors'
+import { Attachment, FileObject } from '~applications/types'
+import IconButton from '@evaka/lib-components/src/atoms/buttons/IconButton'
+import {
+  faExclamationTriangle,
+  faFile,
+  faFileImage,
+  faFilePdf,
+  faFileWord,
+  faTimes
+} from '@evaka/lib-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { UUID } from '~../../lib-common/src/types'
+import { Result } from '~../../lib-common/src/api'
+import { IconDefinition } from '@fortawesome/fontawesome-svg-core'
+
+export type FileUploadProps = {
+  files: Attachment[]
+  onUpload: (
+    file: File,
+    onUploadProgress: (progressEvent: ProgressEvent) => void
+  ) => Promise<Result<UUID>>
+  onDelete: (id: UUID) => Promise<Result<void>>
+}
+
+export type ProgressEvent = {
+  loaded: number
+  total: number
+}
+
+const FileUploadContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+  align-items: flex-start;
+`
+
+const FileInputLabel = styled.label`
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  background: ${colors.greyscale.lightest};
+  border: 1px dashed ${colors.greyscale.medium};
+  border-radius: 8px;
+  width: min(500px, 70vw);
+  padding: 24px;
+  text-align: center;
+
+  & input {
+    display: none;
+  }
+
+  & h4 {
+    font-size: 18px;
+    margin-bottom: 14px;
+  }
+`
+
+const Icon = styled(FontAwesomeIcon)`
+  height: 1rem !important;
+  width: 1rem !important;
+  margin-right: 10px;
+`
+
+const UploadedFiles = styled.div`
+  display: flex;
+  flex-direction: column;
+
+  > *:not(:last-child) {
+    margin-bottom: 20px;
+  }
+`
+
+const File = styled.div`
+  display: flex;
+  flex-direction: row;
+  flex-wrap: nowrap;
+  color: ${colors.greyscale.dark};
+`
+
+const FileIcon = styled(FontAwesomeIcon)`
+  margin-right: 16px;
+  flex: 0;
+  color: ${colors.blues.primary};
+`
+
+const FileDetails = styled.div`
+  flex: 1;
+  display: flex;
+  flex-direction: column;
+  flex-wrap: nowrap;
+`
+
+const FileHeader = styled.div`
+  display: flex;
+  flex-direction: row;
+  flex-wrap: nowrap;
+  justify-content: space-between;
+  font-size: 15px;
```

Comment:
I think File, FileDetails and/or FileHeader might need `align-items: center;` to have everything vertically aligned.